src/data/markdown/docs/30 xk6-browser/01 xk6-browser/02 BrowserContext/pages.md

```diff
@@ -0,0 +1,51 @@
+---
+title: 'pages()'
+excerpt: 'Returns a list of pages inside this BrowserContext.'
+---
+
+<Blockquote mod="warning">
+
+There is a known issue with this feature. See [issue #444](https://github.com/grafana/xk6-browser/issues/444) for details.
+
+</Blockquote>
+
+Returns all open [Page](/javascript-api/xk6-browser/page/)s in the `BrowserContext`.
+
+
+### Returns
+
+| Type   | Description     |
+| ------ | --------------- |
+| array  | All open pages. |
+
+
+### Example
+
+<CodeGroup labels={[]}>
+
+```javascript
+import { chromium } from 'k6/x/browser';
+
+export default function () {
+  const browser = chromium.launch();
+  const context = browser.newContext();
+  context.newPage();
+  const pages = context.pages();
+  console.log(pages); /* [
+                            null,
+                            {
+                                "base_event_emitter": {},
+                                "keyboard": {},
+                                "mouse": {},
+                                "touchscreen": {
+                                    "base_event_emitter": {}
+                                }
+                            }
+                        ] */
+
```

Comment:
Just printing the page length should be enough.

By the way, there is a bug here (other than [#444](https://github.com/grafana/xk6-browser/issues/444)). Sometimes there are two pages and other times three (I ran it multiple times to check for this).

Reply:
Happy to change it to length. @mdcruz what do you think?

Reply:
Resolved in: 7d95d3ebe65fbfeffb588d2e29d38ee795b7f043